beacon_node/lighthouse_network/src/rpc/mod.rs

```diff
@@ -277,6 +378,10 @@ where
         if !self.events.is_empty() {
             return Poll::Ready(self.events.remove(0));
         }
+        if let Poll::Ready(Some(Ok(expired))) = self.next_peer_request.poll_expired(cx) {
+            let (peer_id, protocol) = expired.into_inner();
+            self.send_rate_limited_request(peer_id, protocol)
```

Comment:
I think we might run into issues where, where the `send_rate_limited_request()` adds events to the `self.events` but the task then returns `Poll::Pending` and we won't actually send these events when they are ready until the task wakes up again. 

We should either move this code above the check for `self.events` or check that its not empty again, I think.